src/Exception/RuntimeException.php

```diff
@@ -26,12 +26,11 @@ public static function shouldNotRegister()
         return new self('The current application context does not require a registration');
     }
 
-    public static function requestTypeAlreadyGiven($previous, $type)
+    public static function requestTypeAlreadyKnown($previous)
```

Comment:
Why did you change this? Was the error to specific previously or are you no longer able to get to the newly set type?

Reply:
Yea it was to specific, also wanted to protect against setting the same request type again.